--- a/sympy/concrete/summations.py
+++ b/sympy/concrete/summations.py
@@ -846,11 +846,13 @@
 def eval_sum(f, limits):
     from sympy.concrete.delta import deltasummation, _has_simple_delta
     from sympy.functions import KroneckerDelta
+    from sympy.matrices.expressions.matexpr import MatrixElement
 
     (i, a, b) = limits
     if f is S.Zero:
         return S.Zero
-    if i not in f.free_symbols:
+    # Avoid the constant shortcut for MatrixElement (which may have broken free_symbols)
+    if i not in f.free_symbols and not f.has(MatrixElement):
         return f*(b - a + 1)
     if a == b:
         return f.subs(i, a)
