--- a/django/urls/resolvers.py
+++ b/django/urls/resolvers.py
@@ -250,6 +250,7 @@
         self.converters = _route_to_regex(str(route), is_endpoint)[1]
 
     def match(self, path):
+        from django.http import Http404
         match = self.regex.search(path)
         if match:
             # RoutePattern doesn't allow non-named groups so args are ignored.
@@ -258,7 +259,7 @@
                 converter = self.converters[key]
                 try:
                     kwargs[key] = converter.to_python(value)
-                except ValueError:
+                except (ValueError, Http404):
                     return None
             return path[match.end():], (), kwargs
         return None
