--- a/sympy/printing/glsl.py
+++ b/sympy/printing/glsl.py
@@ -130,6 +130,13 @@
                             A.rows,A.cols)
         elif self._settings['mat_nested']:
             return 'float[%s][%s](\n%s\n)' % (A.rows,A.cols,A.table(self,rowsep=mat_separator,rowstart='float[](',rowend=')'))
+
+    def _print_Identity(self, expr):
+        shape = expr.shape
+        if shape[0].is_Integer and shape[1].is_Integer:
+            return "eye(%s)" % shape[0]
+        else:
+            raise NotImplementedError("Cannot print Identity with symbolic shape")
 
     _print_Matrix = \
         _print_MatrixElement = \
