--- a/sphinx/environment/adapters/indexentries.py
+++ b/sphinx/environment/adapters/indexentries.py
@@ -108,7 +108,8 @@
             lckey = unicodedata.normalize('NFD', key.lower())
             if lckey.startswith('\N{RIGHT-TO-LEFT MARK}'):
                 lckey = lckey[1:]
-            if lckey[0:1].isalpha() or lckey.startswith('_'):
+            if (unicodedata.category(lckey[0:1])[0] == 'L' or
+                    lckey.startswith('_')):
                 lckey = chr(127) + lckey
             # ensure a determinstic order *within* letters by also sorting on
             # the entry itself
@@ -165,7 +166,7 @@
                 if k.startswith('\N{RIGHT-TO-LEFT MARK}'):
                     k = k[1:]
                 letter = unicodedata.normalize('NFD', k[0])[0].upper()
-                if letter.isalpha() or letter == '_':
+                if unicodedata.category(letter)[0] == 'L' or letter == '_':
                     return letter
                 else:
                     # get all other symbols under one heading
