--- a/django/contrib/admin/views/autocomplete.py
+++ b/django/contrib/admin/views/autocomplete.py
@@ -63,6 +63,9 @@
             qs = qs.distinct()
         return qs
 
+    def serialize_result(self, obj, to_field_name):
+        return {'id': str(getattr(obj, to_field_name)), 'text': str(obj)}
+
     def process_request(self, request):
         """
         Validate request integrity, extract and return request parameters.
