--- a/sympy/utilities/lambdify.py
+++ b/sympy/utilities/lambdify.py
@@ -708,7 +708,7 @@
         Returns string form of args, and updated expr.
         """
         from sympy import Dummy, Symbol, Function, flatten
-        from sympy.matrices import DeferredVector
+        from sympy.matrices import DeferredVector, MatrixSymbol
 
         dummify = self._dummify
 
@@ -738,6 +738,14 @@
                 dummy = Dummy()
                 argstrs.append(self._argrepr(dummy))
                 expr = self._subexpr(expr, {arg: dummy})
+            elif isinstance(arg, MatrixSymbol):
+                argrep = self._argrepr(arg)
+                if dummify or not self._is_safe_ident(argrep):
+                    dummy = Dummy()
+                    argstrs.append(self._argrepr(dummy))
+                    expr = self._subexpr(expr, {arg: dummy})
+                else:
+                    argstrs.append(argrep)
             else:
                 argstrs.append(str(arg))
 
