--- a/lib/mpl_toolkits/mplot3d/art3d.py
+++ b/lib/mpl_toolkits/mplot3d/art3d.py
@@ -867,10 +867,16 @@
         self.stale = True
 
     def get_facecolor(self):
-        return self._facecolors2d
+        try:
+            return self._facecolors2d
+        except AttributeError:
+            return super().get_facecolor()
 
     def get_edgecolor(self):
-        return self._edgecolors2d
+        try:
+            return self._edgecolors2d
+        except AttributeError:
+            return super().get_edgecolor()
 
 
 def poly_collection_2d_to_3d(col, zs=0, zdir='z'):
