--- a/django/db/models/sql/compiler.py
+++ b/django/db/models/sql/compiler.py
@@ -1074,7 +1074,7 @@
             _, _, _, joins, _, _ = self.query.setup_joins([f.name], opts, root_alias)
             alias = joins[-1]
             columns = self.get_default_columns(
-                start_alias=alias, opts=f.remote_field.model._meta
+                start_alias=alias, opts=f.remote_field.model._meta.concrete_model._meta
             )
             for col in columns:
                 select_fields.append(len(select))
@@ -1121,7 +1121,7 @@
                 related_klass_infos.append(klass_info)
                 select_fields = []
                 columns = self.get_default_columns(
-                    start_alias=alias, opts=model._meta, from_parent=opts.model
+                    start_alias=alias, opts=model._meta.concrete_model._meta, from_parent=opts.model
                 )
                 for col in columns:
                     select_fields.append(len(select))
@@ -1167,7 +1167,7 @@
                     select_fields = []
                     columns = self.get_default_columns(
                         start_alias=alias,
-                        opts=model._meta,
+                        opts=model._meta.concrete_model._meta,
                         from_parent=opts.model,
                     )
                     for col in columns:
