--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -566,6 +566,7 @@
                 new_model,
                 getattr(old_model._meta, self.option_name, set()),
                 getattr(new_model._meta, self.option_name, set()),
+                from_model=old_model,
             )
 
     def database_backwards(self, app_label, schema_editor, from_state, to_state):
