--- a/sphinx/ext/viewcode.py
+++ b/sphinx/ext/viewcode.py
@@ -186,6 +186,9 @@
     urito = app.builder.get_relative_uri
 
     modnames = set(env._viewcode_modules)  # type: ignore
+
+    if app.builder.name in ('epub', 'epub3') and not app.config.viewcode_enable_epub:
+        return
 
     for modname, entry in status_iterator(
             sorted(env._viewcode_modules.items()),  # type: ignore
