--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -1717,12 +1717,16 @@
 
             if self.objpath[-1] in annotations:
                 objrepr = stringify_typehint(annotations.get(self.objpath[-1]))
+                for alias, aliased in self.config.autodoc_type_aliases.items():
+                    objrepr = re.sub(r'\b%s\b' % re.escape(alias), aliased, objrepr)
                 self.add_line('   :type: ' + objrepr, sourcename)
             else:
                 key = ('.'.join(self.objpath[:-1]), self.objpath[-1])
                 if self.analyzer and key in self.analyzer.annotations:
-                    self.add_line('   :type: ' + self.analyzer.annotations[key],
-                                  sourcename)
+                    type_str = self.analyzer.annotations[key]
+                    for alias, aliased in self.config.autodoc_type_aliases.items():
+                        type_str = re.sub(r'\b%s\b' % re.escape(alias), aliased, type_str)
+                    self.add_line('   :type: ' + type_str, sourcename)
 
             try:
                 if self.object is UNINITIALIZED_ATTR:
@@ -2108,12 +2112,16 @@
 
             if self.objpath[-1] in annotations:
                 objrepr = stringify_typehint(annotations.get(self.objpath[-1]))
+                for alias, aliased in self.config.autodoc_type_aliases.items():
+                    objrepr = re.sub(r'\b%s\b' % re.escape(alias), aliased, objrepr)
                 self.add_line('   :type: ' + objrepr, sourcename)
             else:
                 key = ('.'.join(self.objpath[:-1]), self.objpath[-1])
                 if self.analyzer and key in self.analyzer.annotations:
-                    self.add_line('   :type: ' + self.analyzer.annotations[key],
-                                  sourcename)
+                    type_str = self.analyzer.annotations[key]
+                    for alias, aliased in self.config.autodoc_type_aliases.items():
+                        type_str = re.sub(r'\b%s\b' % re.escape(alias), aliased, type_str)
+                    self.add_line('   :type: ' + type_str, sourcename)
 
             # data descriptors do not have useful values
             if not self._datadescriptor:
