--- a/django/utils/autoreload.py
+++ b/django/utils/autoreload.py
@@ -139,9 +139,9 @@
         path = Path(filename)
         try:
             resolved_path = path.resolve(strict=True).absolute()
-        except FileNotFoundError:
-            # The module could have been removed, don't fail loudly if this
-            # is the case.
+        except (FileNotFoundError, ValueError):
+            # The module could have been removed or the path contains
+            # invalid characters (e.g. null byte), don't fail loudly.
             continue
         results.add(resolved_path)
     return frozenset(results)
