--- a/django/db/models/query_utils.py
+++ b/django/db/models/query_utils.py
@@ -46,10 +46,18 @@
 
         # If the other Q() is empty, ignore it and just use `self`.
         if not other:
-            return copy.deepcopy(self)
+            obj = type(self)()
+            obj.children = self.children[:]
+            obj.connector = self.connector
+            obj.negated = self.negated
+            return obj
         # Or if this Q is empty, ignore it and just use `other`.
         elif not self:
-            return copy.deepcopy(other)
+            obj = type(self)()
+            obj.children = other.children[:]
+            obj.connector = other.connector
+            obj.negated = other.negated
+            return obj
 
         obj = type(self)()
         obj.connector = conn
