--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -758,7 +758,18 @@
                         keep = has_doc or self.options.undoc_members
                 else:
                     keep = False
-            elif (namespace, membername) in attr_docs:
+            if (namespace, membername) in attr_docs:
+                if want_all and isprivate:
+                    if self.options.private_members is None:
+                        keep = False
+                    else:
+                        keep = membername in self.options.private_members
+                else:
+                    # keep documented attributes
+                    keep = True
+                isattr = True
+            elif (isinstance(obj, ObjectMember) and obj.class_ and
+                  (obj.class_.__qualname__, membername) in attr_docs):
                 if want_all and isprivate:
                     if self.options.private_members is None:
                         keep = False
