--- a/sklearn/isotonic.py
+++ b/sklearn/isotonic.py
@@ -4,6 +4,7 @@
 # License: BSD 3 clause
 
 import numpy as np
+from sklearn._config import config_context
 from scipy import interpolate
 from scipy.stats import spearmanr
 from numbers import Real
@@ -410,7 +411,8 @@
         y_pred : ndarray of shape (n_samples,)
             Transformed data.
         """
-        return self.transform(T)
+        with config_context(transform_output="default"):
+            return self.transform(T)
 
     # We implement get_feature_names_out here instead of using
     # `ClassNamePrefixFeaturesOutMixin`` because `input_features` are ignored.
