--- a/django/db/migrations/recorder.py
+++ b/django/db/migrations/recorder.py
@@ -91,13 +91,19 @@
     def record_applied(self, app, name):
         """Record that a migration was applied."""
         self.ensure_schema()
+        if not self._migration_allowed():
+            return
         self.migration_qs.create(app=app, name=name)
 
     def record_unapplied(self, app, name):
         """Record that a migration was unapplied."""
         self.ensure_schema()
+        if not self._migration_allowed():
+            return
         self.migration_qs.filter(app=app, name=name).delete()
 
     def flush(self):
         """Delete all migration records. Useful for testing migrations."""
+        if not self._migration_allowed():
+            return
         self.migration_qs.all().delete()
