--- a/django/core/files/storage.py
+++ b/django/core/files/storage.py
@@ -214,7 +214,10 @@
 
     @cached_property
     def file_permissions_mode(self):
-        return self._value_or_setting(self._file_permissions_mode, settings.FILE_UPLOAD_PERMISSIONS)
+        mode = self._value_or_setting(self._file_permissions_mode, settings.FILE_UPLOAD_PERMISSIONS)
+        if mode is None:
+            return 0o644
+        return mode
 
     @cached_property
     def directory_permissions_mode(self):
