--- a/django/forms/widgets.py
+++ b/django/forms/widgets.py
@@ -59,21 +59,25 @@
 
     @property
     def _css(self):
-        css = self._css_lists[0]
+        css = {}
         # filter(None, ...) avoids calling merge with empty dicts.
-        for obj in filter(None, self._css_lists[1:]):
-            css = {
-                medium: self.merge(css.get(medium, []), obj.get(medium, []))
-                for medium in css.keys() | obj.keys()
-            }
+        # The order of the lists is important, so we can't use a set.
+        # Prioritize later-added lists by merging them into the current result.
+        for obj in filter(None, self._css_lists):
+            new_css = {}
+            for medium in obj.keys() | css.keys():
+                new_css[medium] = self.merge(obj.get(medium, []), css.get(medium, []))
+            css = new_css
         return css
 
     @property
     def _js(self):
-        js = self._js_lists[0]
+        js = []
         # filter(None, ...) avoids calling merge() with empty lists.
-        for obj in filter(None, self._js_lists[1:]):
-            js = self.merge(js, obj)
+        # The order of the lists is important, so we can't use a set.
+        # Prioritize later-added lists by merging them into the current result.
+        for obj in filter(None, self._js_lists):
+            js = self.merge(obj, js)
         return js
 
     def render(self):
