--- a/django/test/utils.py
+++ b/django/test/utils.py
@@ -180,7 +180,7 @@
                         verbosity=verbosity,
                         autoclobber=not interactive,
                         keepdb=keepdb,
-                        serialize=connection.settings_dict['TEST'].get('SERIALIZE', True),
+                        serialize=connection.settings_dict['TEST'].get('SERIALIZE', connection.settings_dict['TEST'].get('MIGRATE', True)),
                     )
                 if parallel > 1:
                     for index in range(parallel):
