--- a/sympy/polys/polyoptions.py
+++ b/sympy/polys/polyoptions.py
@@ -457,10 +457,8 @@
 
                 gens = list(map(sympify, gens.split(',')))
 
-                if ground in ['Z', 'ZZ']:
-                    return sympy.polys.domains.ZZ.poly_ring(*gens)
-                else:
-                    return sympy.polys.domains.QQ.poly_ring(*gens)
+                ground_domain = cls.preprocess(ground)
+                return ground_domain.poly_ring(*gens)
 
             r = cls._re_fraction.match(domain)
 
@@ -469,10 +467,8 @@
 
                 gens = list(map(sympify, gens.split(',')))
 
-                if ground in ['Z', 'ZZ']:
-                    return sympy.polys.domains.ZZ.frac_field(*gens)
-                else:
-                    return sympy.polys.domains.QQ.frac_field(*gens)
+                ground_domain = cls.preprocess(ground)
+                return ground_domain.frac_field(*gens)
 
             r = cls._re_algebraic.match(domain)
 
@@ -480,7 +476,7 @@
                 gens = list(map(sympify, r.groups()[1].split(',')))
                 return sympy.polys.domains.QQ.algebraic_field(*gens)
 
-        raise OptionError('expected a valid domain specification, got %s' % domain)
+        raise OptionError("expected a valid domain specification, e.g. 'ZZ[x]', 'QQ[x]', 'RR[x]' or 'CC[x]', got %s" % domain)
 
     @classmethod
     def postprocess(cls, options):
