--- a/sympy/polys/ring_series.py
+++ b/sympy/polys/ring_series.py
@@ -233,7 +233,7 @@
         raise ValueError('p1 and p2 must have the same ring')
     iv = R.gens.index(x)
     if not isinstance(p2, PolyElement):
-        raise ValueError('p1 and p2 must have the same ring')
+        raise ValueError('p2 must be a polynomial')
     if R == p2.ring:
         get = p.get
         items2 = list(p2.items())
