--- a/sympy/parsing/mathematica.py
+++ b/sympy/parsing/mathematica.py
@@ -936,7 +936,7 @@
                     args = [converter(arg) for arg in expr[1:]]
                     return Function(head)(*args)
                 else:
-                    raise ValueError("error")
+                    raise ValueError("Received an empty list of expressions")
             elif isinstance(expr, str):
                 return Symbol(expr)
             else:
