--- a/sphinx/writers/html5.py
+++ b/sphinx/writers/html5.py
@@ -297,6 +297,9 @@
 
         figtype = self.builder.env.domains['std'].get_enumerable_node_type(node)
         if figtype:
+            title = self.builder.env.domains['std'].get_numfig_title(node)
+            if title is None:
+                return  # no title, no numbering
             if len(node['ids']) == 0:
                 msg = __('Any IDs not assigned for %s node') % node.tagname
                 logger.warning(msg, location=node)
