--- a/django/contrib/contenttypes/models.py
+++ b/django/contrib/contenttypes/models.py
@@ -92,7 +92,7 @@
                 self._add_to_cache(self.db, ct)
         # Create content types that weren't in the cache or DB.
         for opts, opts_models in needed_opts.items():
-            ct = self.create(
+            ct, created = self.get_or_create(
                 app_label=opts.app_label,
                 model=opts.model_name,
             )
