--- a/sympy/core/numbers.py
+++ b/sympy/core/numbers.py
@@ -1359,10 +1359,10 @@
                   -> p**r*(sin(Pi*r) + cos(Pi*r)*I)
         """
         if self == 0:
-            if expt.is_positive:
+            if expt.is_extended_positive:
                 return S.Zero
-            if expt.is_negative:
-                return S.Infinity
+            if expt.is_extended_negative:
+                return S.ComplexInfinity
         if isinstance(expt, Number):
             if isinstance(expt, Integer):
                 prec = self._prec
@@ -2623,9 +2623,9 @@
         return S.Zero
 
     def _eval_power(self, expt):
-        if expt.is_positive:
+        if expt.is_extended_positive:
             return self
-        if expt.is_negative:
+        if expt.is_extended_negative:
             return S.ComplexInfinity
         if expt.is_extended_real is False:
             return S.NaN
