--- a/sympy/simplify/simplify.py
+++ b/sympy/simplify/simplify.py
@@ -562,7 +562,7 @@
 
     def done(e):
         rv = e.doit() if doit else e
-        return shorter(rv, collect_abs(rv))
+        return shorter(rv, rv)
 
     expr = sympify(expr)
     kwargs = dict(
@@ -579,7 +579,7 @@
     if _eval_simplify is not None:
         return _eval_simplify(**kwargs)
 
-    original_expr = expr = collect_abs(signsimp(expr))
+    original_expr = expr = signsimp(expr)
 
     if not isinstance(expr, Basic) or not expr.args:  # XXX: temporary hack
         return expr
