--- a/django/contrib/admin/options.py
+++ b/django/contrib/admin/options.py
@@ -2037,10 +2037,14 @@
         self.opts = self.model._meta
         self.has_registered_model = admin_site.is_registered(self.model)
         super().__init__()
+        explicit_verbose_name = self.verbose_name is not None
         if self.verbose_name is None:
             self.verbose_name = self.model._meta.verbose_name
         if self.verbose_name_plural is None:
-            self.verbose_name_plural = self.model._meta.verbose_name_plural
+            if explicit_verbose_name:
+                self.verbose_name_plural = self.verbose_name + 's'
+            else:
+                self.verbose_name_plural = self.model._meta.verbose_name_plural
 
     @property
     def media(self):
