--- a/django/db/models/fields/__init__.py
+++ b/django/db/models/fields/__init__.py
@@ -562,6 +562,60 @@
         return _load_field, (self.model._meta.app_label, self.model._meta.object_name,
                              self.name)
 
+    def deconstruct(self):
+        """
+        Return a 3-tuple of class import path, positional arguments, and keyword
+        arguments.
+        """
+        path = '%s.%s' % (self.__class__.__module__, self.__class__.__qualname__)
+        args = []
+        kwargs = {}
+        if self.name is not None:
+            kwargs['name'] = self.name
+        if self.verbose_name is not None and self.verbose_name != self.name:
+            kwargs['verbose_name'] = self.verbose_name
+        if self.primary_key:
+            kwargs['primary_key'] = True
+        if self.max_length is not None:
+            kwargs['max_length'] = self.max_length
+        if self._unique:
+            kwargs['unique'] = True
+        if self.blank:
+            kwargs['blank'] = True
+        if self.null:
+            kwargs['null'] = True
+        if self.db_index is not None:
+            kwargs['db_index'] = self.db_index
+        if self.remote_field:
+            kwargs['rel'] = self.remote_field
+        if self.default is not NOT_PROVIDED:
+            kwargs['default'] = self.default
+        if self.editable is not True:
+            kwargs['editable'] = False
+        if self.serialize is not True:
+            kwargs['serialize'] = False
+        if self.unique_for_date is not None:
+            kwargs['unique_for_date'] = self.unique_for_date
+        if self.unique_for_month is not None:
+            kwargs['unique_for_month'] = self.unique_for_month
+        if self.unique_for_year is not None:
+            kwargs['unique_for_year'] = self.unique_for_year
+        if self.choices:
+            kwargs['choices'] = self.choices
+        if self.help_text:
+            kwargs['help_text'] = self.help_text
+        if self.db_column is not None:
+            kwargs['db_column'] = self.db_column
+        if self.db_tablespace is not None:
+            kwargs['db_tablespace'] = self.db_tablespace
+        if self.auto_created:
+            kwargs['auto_created'] = True
+        if self._validators:
+            kwargs['validators'] = self._validators
+        if self._error_messages:
+            kwargs['error_messages'] = self._error_messages
+        return path, args, kwargs
+
     def get_pk_value_on_save(self, instance):
         """
         Hook to generate new PK values on save. This method is called when
