--- a/django/utils/decorators.py
+++ b/django/utils/decorators.py
@@ -60,7 +60,7 @@
     # of the method that will be decorated.
     def _dec(obj):
         if not isinstance(obj, type):
-            return _multi_decorate(decorator, obj)
+            return update_wrapper(_multi_decorate(decorator, obj), obj)
         if not (name and hasattr(obj, name)):
             raise ValueError(
                 "The keyword argument `name` must be the name of a method "
@@ -73,6 +73,7 @@
                 "%s (%s)." % (name, obj, method)
             )
         _wrapper = _multi_decorate(decorator, method)
+        _wrapper = update_wrapper(_wrapper, method)
         setattr(obj, name, _wrapper)
         return obj
 
