--- a/sympy/functions/elementary/exponential.py
+++ b/sympy/functions/elementary/exponential.py
@@ -447,6 +447,13 @@
         from sympy import cos
         I = S.ImaginaryUnit
         return cos(I*arg) + I*cos(I*arg + S.Pi/2)
+
+    def _eval_trigsimp(self, **opts):
+        from sympy import sin
+        arg = self.args[0]
+        if arg.has(S.ImaginaryUnit):
+            return self.rewrite(sin)
+        return self
 
     def _eval_rewrite_as_tanh(self, arg):
         from sympy import tanh
