--- a/sympy/printing/latex.py
+++ b/sympy/printing/latex.py
@@ -1968,9 +1968,12 @@
             tex = r"\left(%s\right)^{%s}" % (tex, exp)
         return tex
 
-    def _print_SingularityFunction(self, expr):
+    def _print_SingularityFunction(self, expr, exp=None):
         shift = self._print(expr.args[0] - expr.args[1])
-        power = self._print(expr.args[2])
+        if exp is None:
+            power = self._print(expr.args[2])
+        else:
+            power = exp
         tex = r"{\left\langle %s \right\rangle}^{%s}" % (shift, power)
         return tex
 
