--- a/sympy/assumptions/refine.py
+++ b/sympy/assumptions/refine.py
@@ -64,6 +64,19 @@
     if not isinstance(new_expr, Expr):
         return new_expr
     return refine(new_expr, assumptions)
+
+def refine_arg(expr, assumptions):
+    from sympy import ask, Q, S
+    arg = expr.args[0]
+    if ask(Q.real(arg) & Q.positive(arg), assumptions):
+        return S.Zero
+    if ask(Q.real(arg) & Q.negative(arg), assumptions):
+        return S.Pi
+    if ask(Q.zero(arg), assumptions):
+        return S.Zero
+    return expr
+
+handlers_dict['arg'] = refine_arg
 
 
 def refine_abs(expr, assumptions):
