--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -937,9 +937,10 @@
                     # Use pk from related object if it has been saved after
                     # an assignment.
                     setattr(self, field.attname, obj.pk)
-                # If the relationship's pk/to_field was changed, clear the
-                # cached relationship.
+                # If the relationship's pk/to_field was changed, update the
+                # FK field and clear the cached relationship.
                 if getattr(obj, field.target_field.attname) != getattr(self, field.attname):
+                    setattr(self, field.attname, obj.pk)
                     field.delete_cached_value(self)
 
     def delete(self, using=None, keep_parents=False):
