--- a/sklearn/cluster/_affinity_propagation.py
+++ b/sklearn/cluster/_affinity_propagation.py
@@ -200,6 +200,9 @@
     else:
         if verbose:
             print("Did not converge")
+        warnings.warn("Affinity propagation did not converge, this model "
+                      "will not have any cluster centers.", ConvergenceWarning)
+        return (np.array([], dtype=int), np.array([-1] * n_samples), max_iter) if return_n_iter else (np.array([], dtype=int), np.array([-1] * n_samples))
 
     I = np.flatnonzero(E)
     K = I.size  # Identify exemplars
